pkg/oci/build_metadata.go

```diff
@@ -129,7 +129,7 @@ func createOrUpdateMetadataFile(ctx context.Context, opts *BuildGadgetImageOpts)
 
 	// fix owner of created metadata file
 	if !update {
-		if err := fixMetadataOwner(opts); err != nil {
+		if err := fixOwner(opts.MetadataPath, opts.EBPFSourcePath); err != nil {
```

Comment:
We need to use separate files for each OS and use build tags like `//go:build`, otherwise, the compilation fails:
```
$ make gadgetctl-windows-amd64
... undefined: syscall.Stat_t
```